### PR TITLE
Return and Space keys do not trigger onPress events

### PR DESCRIPTION
**Pull request by tom-un**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [X] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

This is a fix for Issue [1622](https://github.com/microsoft/react-native-macos/issues/1622) .  This fixes a regression that was introduced in .68 when the `validKeysDown`/`validKeysUp` View props were introduced.  Since then various Pressability components (Pressable, and Touchable*) do not trigger `onPress` events when the Return or Spacebar key is pressed.

The fix involves triggering the `onPressIn` and `onPressOut` events in `Pressibility`'s `onKeyDown`/`onKeyUp` handlers similar to how `react-native-windows` also does this: https://github.com/microsoft/react-native-windows/blob/main/vnext/src/Libraries/Pressability/Pressability.windows.js.  It also triggers the `onPress` on key down which mimics how most macOS controls respond to key down as a click in contrast to Windows which triggers `onPress` on key up.

Logic was also added to `RCTView`'s `keyboardEvent:` filtering logic: if the view is focusable and it does not have `validKeyDown/Up` props, it implicitly allows the Return and Spacebar keys.

## Changelog

[macOS] [Fix] - Return and Space keys do not trigger onPress events

## Test Plan

All the Pressable and Touchable pages in RNTester were tested:
Tom branch

Note that this change is very similar to Saad's PR here: https://github.com/microsoft/react-native-macos/pull/1614.  The changes to `Pressability` event handlers are identical in both PRs.  But in Saad's the `validKeysDown` prop is giving a default value of Enter and Space in the Pressability.js while this PR makes the default values of the `validKeysDown/Up` in the native `RCTView`. The advantage of doing it in the native side means that the Touchable* components also get the correct behavior without making very large scale changes to the various Touchable* files -- Saad's change only fixes Pressable.  Also the other PR only makes default values for `validKeysDown` and not also `validKeysUp`, so the Press Out effects never trigger leaving the component in a "stuck" state.  Here is a sample of the Pressable test in the other branch:

Saad branch

### CODE REVIEW

File: React/Views/RCTView.m [hunk 1]
```diff
@@ -1667,6 +1667,12 @@ - (RCTViewKeyboardEvent*)keyboardEvent:(NSEvent*)event {
   NSArray<NSString *> *validKeys = keyDown ? self.validKeysDown : self.validKeysUp;
   NSString *key = [RCTViewKeyboardEvent keyFromEvent:event];
 
+  // If the view is focusable and the component didn't explicity set the validKeysDown or Up,
+  // allow enter/return and spacebar key events to mimic the behavior of native controls.
+  if (self.focusable && validKeys == nil) {
+    validKeys = @[@"Enter", @" "];
```
Comment (mischreiber) [inline React/Views/RCTView.m:1673]: Does this include both the Return and Enter keys on a Mac keyboard? They're technically different key codes.

Comment (Saadnajmi) [reply React/Views/RCTView.m:1673]: From `RCTViewKeyboardEvent.m`

```
  } else if (code == NSEnterCharacter || code == NSNewlineCharacter || code == NSCarriageReturnCharacter) {
    return @"Enter";
```

Comment (Saadnajmi) [reply React/Views/RCTView.m:1673]: I'm pretty sure the w3c web spec is that the `key` value of this event should say "Enter" : https://www.w3.org/TR/uievents-key/#keys-whitespace

That's what we follow for React Native Desktop.

Comment (mischreiber) [reply React/Views/RCTView.m:1673]: Cool, thanks!

File: Libraries/Pressability/Pressability.js [hunk 2]
```diff
@@ -628,12 +628,35 @@ export default class Pressability {
...
+        if (
+          (event.nativeEvent.key === 'Enter' ||
+            event.nativeEvent.key === ' ') &&
+          event.defaultPrevented !== true
+        ) {
+          const {onPress, onPressIn} = this._config;
+          // $FlowFixMe: PressEvents don't mesh with keyboarding APIs. Keep legacy behavior of passing KeyEvents instead
+          onPress && onPress(event);
+          // $FlowFixMe: PressEvents don't mesh with keyboarding APIs. Keep legacy behavior of passing KeyEvents instead
+          onPressIn && onPressIn(event);
```
Comment (Saadnajmi) [inline Libraries/Pressability/Pressability.js:642]: onPress should be called after onPressIn I think?

**Review by Saadnajmi — APPROVED** [review]

**Review by Saadnajmi — APPROVED** [review]
Long term I'd still rather the logic live in JS like in #1614 . However, given that change needs more work and this affects people _now_, I say we merge/backport this version. If/when #1615 lands, this logic would change anyway.

**Pull request merged**